src/nimony/sem.nim

```diff
@@ -616,13 +616,17 @@ proc semConstExpr(c: var SemContext; it: var Item) =
 
 proc semStmtsExpr(c: var SemContext; it: var Item; isNewScope: bool) =
   let before = c.dest.len
+  if isNewScope:
+    openScope c
```

Comment:
Should not be necessary as `sem` generates `ScopeS` nodes, but does not depend on them for scope creation. Instead all the control flow constructs already know when they need to introduce new scopes.

Reply:
Here I am building `(scope)` for each unroll of the loop before passing it to typechecking, I can replace it with `if true` but I figured this might be less complex. `block` would interfere with `break`.